src/java.base/share/classes/java/lang/classfile/ClassFile.java

```diff
@@ -441,967 +484,967 @@ default byte[] transform(ClassModel model, ClassDesc newClassName, ClassTransfor
 
-    /** 0 */
+    /** The value of verification type TOP. */
     int VT_TOP = 0;
 
-    /** 1 */
+    /** The value of verification type INTEGER. */
     int VT_INTEGER = 1;
 
-    /** 2 */
+    /** The value of verification type FLOAT. */
     int VT_FLOAT = 2;
 
-    /** 3 */
+    /** The value of verification type DOUBLE. */
     int VT_DOUBLE = 3;
 
-    /** 4 */
+    /** The value of verification type LONG. */
     int VT_LONG = 4;
 
-    /** 5 */
+    /** The value of verification type NULL. */
     int VT_NULL = 5;
 
-    /** 6 */
+    /** The value of verification type UNINITIALIZED_THIS. */
     int VT_UNINITIALIZED_THIS = 6;
 
-    /** 7 */
+    /** The value of verification type OBJECT. */
     int VT_OBJECT = 7;
 
-    /** 8 */
+    /** The value of verification type UNINITIALIZED. */
     int VT_UNINITIALIZED = 8;
 
-    /** ACC_PUBLIC */
+    /** The value of default class access flags */
     int DEFAULT_CLASS_FLAGS = ACC_PUBLIC;
 
-    /** 45 */
+    /** The class major version of JAVA_1. */
     int JAVA_1_VERSION = 45;
 
-    /** 46 */
+    /** The class major version of JAVA_2. */
     int JAVA_2_VERSION = 46;
 
-    /** 47 */
+    /** The class major version of JAVA_3. */
     int JAVA_3_VERSION = 47;
 
-    /** 48 */
+    /** The class major version of JAVA_4. */
     int JAVA_4_VERSION = 48;
 
-    /** 49 */
+    /** The class major version of JAVA_5. */
     int JAVA_5_VERSION = 49;
 
-    /** 50 */
+    /** The class major version of JAVA_6. */
     int JAVA_6_VERSION = 50;
 
-    /** 51 */
+    /** The class major version of JAVA_7. */
     int JAVA_7_VERSION = 51;
 
-    /** 52 */
+    /** The class major version of JAVA_8. */
     int JAVA_8_VERSION = 52;
 
-    /** 53 */
+    /** The class major version of JAVA_9. */
     int JAVA_9_VERSION = 53;
 
-    /** 54 */
+    /** The class major version of JAVA_10. */
     int JAVA_10_VERSION = 54;
 
-    /** 55 */
+    /** The class major version of JAVA_11. */
     int JAVA_11_VERSION = 55;
 
-    /** 56 */
+    /** The class major version of JAVA_12. */
     int JAVA_12_VERSION = 56;
 
-    /** 57 */
+    /** The class major version of JAVA_13. */
     int JAVA_13_VERSION = 57;
 
-    /** 58 */
+    /** The class major version of JAVA_14. */
     int JAVA_14_VERSION = 58;
 
-    /** 59 */
+    /** The class major version of JAVA_15. */
     int JAVA_15_VERSION = 59;
 
-    /** 60 */
+    /** The class major version of JAVA_16. */
     int JAVA_16_VERSION = 60;
 
-    /** 61 */
+    /** The class major version of JAVA_17. */
     int JAVA_17_VERSION = 61;
 
-    /** 62 */
+    /** The class major version of JAVA_18. */
     int JAVA_18_VERSION = 62;
 
-    /** 63 */
+    /** The class major version of JAVA_19. */
     int JAVA_19_VERSION = 63;
 
-    /** 64 */
+    /** The class major version of JAVA_20. */
     int JAVA_20_VERSION = 64;
 
-    /** 65 */
+    /** The class major version of JAVA_21. */
     int JAVA_21_VERSION = 65;
 
-    /** 66 */
```

Comment:
Is there a more scalable way to express the major class version? The current approach means we have to add two fields per year.  Maybe a lookup function would be better?

Reply:
This is explicit list of supported class file versions, so I don't see any other way.

Reply:
Would it not be possible to expose an immutable `Map<Integer, Integer>` that maps from Java version to major class version?